jiant/utils/utils.py

```diff
@@ -361,15 +401,40 @@ def get_elmo_mixing_weights(text_field_embedder, task=None):
     return params
 
 
-def get_batch_size(batch, keyword="input"):
+def format_output(obj, cuda_devices):
+    """
+    Format output based on whether model is using DataParallel or not. 
+    DataParallel necessitates objects to be gathered into GPU:0 to have 
+    dimension 0. 
+    This function will be used for scalar outputs of model forwards 
+    such as loss and n_exs.
+    """
+    if isinstance(cuda_devices, list):
+        if isinstance(obj, torch.Tensor) is False:
```

Comment:
`if not isinstance(obj, torch.Tensor)` is more pythonic

Reply:
what type would `obj` by in this case ?

Reply:
It could be a scalar for example and not a torch.Tensor. For example, out["loss"] is usually passed into format_output

Reply:
Because out["loss"] is already on cuda. We just explicitly put cuda here because we're creating a torch.Tensor.